src/test/java/com/algotrading/backtesting/replay/test/ActiveTradedStocksTest.java

```diff
@@ -0,0 +1,77 @@
+
+import com.algotrading.backtesting.replay.ActiveTradedStocksReader;
+import com.algotrading.backtesting.replay.ActivelyTradedStocks;
+import com.algotrading.backtesting.replay.AvailableStocks;
+import com.algotrading.backtesting.stock.Stock;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.*;
+
+import static com.algotrading.constants.TestConstants.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ActiveTradedStocksTest {
+
+    private static final SimpleDateFormat sdf = new SimpleDateFormat("yyyyMMdd");
+
+
+
+    public static final Date DATE_20130101 = new GregorianCalendar(2013, 0, 1).getTime(); // 20130101
+    public static final Date DATE_20140101 = new GregorianCalendar(2014, 0, 1).getTime(); // 20140101
+    public static final Date DATE_20150101 = new GregorianCalendar(2015, 0, 1).getTime(); // 20150101
+    @Mock
+    private ActiveTradedStocksReader activeTradedStocksReader;
+
+    @Before
+    public void setup() throws IOException {
+        Map<String, List<String>> map = new HashMap<>();
+        map.put(sdf.format(DATE_20130101), Arrays.asList(HK_0001, HK_0002, HK_0003, HK_0004, HK_0005));
+        map.put(sdf.format(DATE_20140101), Arrays.asList(HK_0006, HK_0007, HK_0008));
+        map.put(sdf.format(DATE_20150101), Arrays.asList(HK_0009, HK_0010, HK_0011, HK_0012));
+        Mockito.when(activeTradedStocksReader.constructYearToTickerListFromFiles()).thenReturn(map);
+    }
+
+    @Test
+    public void testAbsolute() throws IOException {
+        ActivelyTradedStocks activelyTradedStocks = new ActivelyTradedStocks(activeTradedStocksReader, 3, true);
+        Map<String, Stock> allAvailableStocks = activelyTradedStocks.getAllAvailableStocks();
+        Assert.assertEquals(9, allAvailableStocks.size());
+        Assert.assertEquals(HK_0001, allAvailableStocks.get(HK_0001).getTicker());
+        Assert.assertEquals(HK_0002, allAvailableStocks.get(HK_0002).getTicker());
+        Assert.assertEquals(HK_0003, allAvailableStocks.get(HK_0003).getTicker());
+
+        Assert.assertEquals(HK_0006, allAvailableStocks.get(HK_0006).getTicker());
+        Assert.assertEquals(HK_0007, allAvailableStocks.get(HK_0007).getTicker());
+        Assert.assertEquals(HK_0008, allAvailableStocks.get(HK_0008).getTicker());
+
+        Assert.assertEquals(HK_0009, allAvailableStocks.get(HK_0009).getTicker());
+        Assert.assertEquals(HK_0010, allAvailableStocks.get(HK_0010).getTicker());
+        Assert.assertEquals(HK_0011, allAvailableStocks.get(HK_0011).getTicker());
+
+        AvailableStocks availableStocks2013 = activelyTradedStocks.get(DATE_20130101);
+        Assert.assertNull(availableStocks2013); // I am not sure is it intented, in implementation only "dateStr.compareTo(d) > 0"
```

Comment:
On 20130101, should switch to the list in 20130101.txt. It should be ">=" instead of ">"